client/src/app/components/users/users.component.html

```diff
@@ -4,20 +4,20 @@
     <button mat-raised-button color="primary">Add User</button>
   </div>
 
-  <table mat-table [dataSource]="users" class="mat-elevation-z8">
+  <table mat-table [dataSource]="users" matSort class="mat-elevation-z8">
```

Comment:
It looks like `matSort` directive here also needs `matSortChange` to be set to a method defined in the `.ts.` file: `(matSortChange)='mySortingMethod($event)'`